### PR TITLE
Move AzureML-MLDesigner to registry

### PR DESCRIPTION
**Pull request by zhengfeiwang**
Move pipelines team owned curated environment `AzureML-MLDesigner` to this repo, split it into two environments: `mldesigner-2.0-ubuntu20.04-py38-cpu` and `mldesigner-minimal-2.0-ubuntu20.04-py38-cpu`.

### CODE REVIEW

File: pipelines/environments/componentsdk-1.5-ubuntu20.04-py38-cpu/context/Dockerfile
Comment (elesel) [inline pipelines/environments/componentsdk-1.5-ubuntu20.04-py38-cpu/context/Dockerfile:13]: Unrelated comment.
```suggestion
```

File: pipelines/environments/componentsdk-1.5-ubuntu20.04-py38-cpu/asset.yaml
Comment (elesel) [inline pipelines/environments/componentsdk-1.5-ubuntu20.04-py38-cpu/asset.yaml:1]: Do you plan to use automated versioning? If so, you should include `version: auto` See https://github.com/Azure/azureml-assets/wiki/Environments#assetyaml

File: pipelines/environments/mldesigner-2.0-ubuntu20.04-py38-cpu/asset.yaml
Comment (elesel) [inline pipelines/environments/mldesigner-2.0-ubuntu20.04-py38-cpu/asset.yaml:1]: Do you plan to use automated versioning? If so, you should include `version: auto` See https://github.com/Azure/azureml-assets/wiki/Environments#assetyaml

File: pipelines/environments/mldesigner-2.0-ubuntu20.04-py38-cpu/context/Dockerfile
Comment (elesel) [inline pipelines/environments/mldesigner-2.0-ubuntu20.04-py38-cpu/context/Dockerfile:14]: Please pin the `mldesigner` version by using `=={{latest-pypi-version}}` or hard-coding it here. This will help reproducibility.

File: pipelines/environments/componentsdk-1.5-ubuntu20.04-py38-cpu/context/Dockerfile
Comment (elesel) [inline pipelines/environments/componentsdk-1.5-ubuntu20.04-py38-cpu/context/Dockerfile:6]: Consider using a conda.yaml file. Here's an example: https://github.com/Azure/azureml-assets/blob/main/training/general/src/environments/sklearn-1.1-ubuntu20.04-py38-cpu/context/Dockerfile

File: pipelines/environments/mldesigner-2.0-ubuntu20.04-py38-cpu/context/Dockerfile
Comment (elesel) [inline pipelines/environments/mldesigner-2.0-ubuntu20.04-py38-cpu/context/Dockerfile:6]: Consider using a conda.yaml file. Here's an example: https://github.com/Azure/azureml-assets/blob/main/training/general/src/environments/sklearn-1.1-ubuntu20.04-py38-cpu/context/Dockerfile

File: pipelines/environments/componentsdk-1.5-ubuntu20.04-py38-cpu/asset.yaml
Comment (elesel) [inline pipelines/environments/componentsdk-1.5-ubuntu20.04-py38-cpu/asset.yaml:4]: Please add newlines at the end of your files.

**Review by elesel — CHANGES_REQUESTED** [review]

File: pipelines/environments/componentsdk-1.5-ubuntu20.04-py38-cpu/context/Dockerfile
Comment (zhengfeiwang) [reply pipelines/environments/componentsdk-1.5-ubuntu20.04-py38-cpu/context/Dockerfile:13]: Forget to delete this, will remove this line.

File: pipelines/environments/componentsdk-1.5-ubuntu20.04-py38-cpu/asset.yaml
Comment (zhengfeiwang) [reply pipelines/environments/componentsdk-1.5-ubuntu20.04-py38-cpu/asset.yaml:1]: Yes, and added `version: auto` for this, thanks for pointing out!

File: pipelines/environments/mldesigner-2.0-ubuntu20.04-py38-cpu/asset.yaml
Comment (zhengfeiwang) [reply pipelines/environments/mldesigner-2.0-ubuntu20.04-py38-cpu/asset.yaml:1]: Yes, and added `version: auto` for this, thanks for pointing out!

File: pipelines/environments/componentsdk-1.5-ubuntu20.04-py38-cpu/asset.yaml
Comment (zhengfeiwang) [reply pipelines/environments/componentsdk-1.5-ubuntu20.04-py38-cpu/asset.yaml:4]: Added for all files in my branch.

File: pipelines/environments/mldesigner-2.0-ubuntu20.04-py38-cpu/context/Dockerfile
Comment (zhengfeiwang) [reply pipelines/environments/mldesigner-2.0-ubuntu20.04-py38-cpu/context/Dockerfile]: Hard-code this for now

File: pipelines/environments/componentsdk-1.5-ubuntu20.04-py38-cpu/context/Dockerfile
Comment (zhengfeiwang) [reply pipelines/environments/componentsdk-1.5-ubuntu20.04-py38-cpu/context/Dockerfile]: Have updated with using conda.yaml

File: pipelines/environments/mldesigner-2.0-ubuntu20.04-py38-cpu/context/Dockerfile
Comment (zhengfeiwang) [reply pipelines/environments/mldesigner-2.0-ubuntu20.04-py38-cpu/context/Dockerfile]: Have updated with using conda.yaml

**Review by 0mza987 — DISMISSED** [review]

File: pipelines/environments/mldesigner-2.0-ubuntu20.04-py38-cpu/asset.yaml
Comment (elesel) [inline pipelines/environments/mldesigner-2.0-ubuntu20.04-py38-cpu/asset.yaml:1]: Would `mldesigner` work for you as the environment name, if the `2.0` designation isn't important? We're trying to lean toward simpler environment names where possible. See https://github.com/Azure/azureml-assets/wiki/Asset-naming-convention#environments
```suggestion
name: mldesigner
```

File: pipelines/environments/mldesigner-minimal-2.0-ubuntu20.04-py38-cpu copy/asset.yaml
Comment (elesel) [inline pipelines/environments/mldesigner-minimal-2.0-ubuntu20.04-py38-cpu copy/asset.yaml:1]:
```suggestion
name: mldesigner-minimal
```

File: pipelines/environments/mldesigner-minimal-2.0-ubuntu20.04-py38-cpu copy/context/conda_dependencies.yaml
Comment (elesel) [inline pipelines/environments/mldesigner-minimal-2.0-ubuntu20.04-py38-cpu copy/context/conda_dependencies.yaml:9]: If the only difference between this environment and the non-minimal version is azure-ai-ml and azureml-mlflow, does this environment need to exist?

File: pipelines/environments/mldesigner-minimal-2.0-ubuntu20.04-py38-cpu copy/environment.yaml
Comment (elesel) [inline pipelines/environments/mldesigner-minimal-2.0-ubuntu20.04-py38-cpu copy/environment.yaml:2]: Is this in the MCR config? I only see azureml/curated/mldesigner there right now.

**Review by elesel — CHANGES_REQUESTED** [review]

File: pipelines/environments/mldesigner-minimal-2.0-ubuntu20.04-py38-cpu copy/context/conda_dependencies.yaml
Comment (zhengfeiwang) [reply pipelines/environments/mldesigner-minimal-2.0-ubuntu20.04-py38-cpu copy/context/conda_dependencies.yaml:9]: Yes, `mldesigner` can work without `azure-ai-ml` and `azureml-mlflow`. and we recommend users to only use `mldesigner` if they do not require any advanced features in a command component.

File: pipelines/environments/mldesigner-minimal-2.0-ubuntu20.04-py38-cpu copy/environment.yaml
Comment (zhengfeiwang) [reply pipelines/environments/mldesigner-minimal-2.0-ubuntu20.04-py38-cpu copy/environment.yaml:2]: We have a PR working on this: microsoft/mcr#2383

Comment (elesel) [reply pipelines/environments/mldesigner-minimal-2.0-ubuntu20.04-py38-cpu copy/environment.yaml:2]: We'll need to wait for the MCR PR to merge before this PR can merge.

**Review by elesel — APPROVED** [review]

**Pull request merged**
